Firestore/core/test/firebase/firestore/util/status_test.cc

```diff
@@ -55,13 +55,35 @@ TEST(Status, Copy) {
   ASSERT_EQ(a.ToString(), b.ToString());
 }
 
+TEST(Status, Move) {
+  Status s(Status(Error::InvalidArgument, "Invalid"));
```

Comment:
Can you add some more tests here?
* Check that invoking move assignment operator is okay (the current test only exercises the move constructor), with both error and non-error statuses:
```cpp
{
  Status ok;
  Status reassigned1{Error::InvalidArgument, "Foo"};
  reassigned = std::move(ok);
   // check the state of reassigned...
}
{
  Status bad{Error::InvalidArgument, "Foo"};
  Status reassigned1{Error::ResourceExhausted, "Bar"};
  reassigned = std::move(bad);
   // check the state of reassigned...
}
```
* check the pathological case where a status is move-assigned to itself:
```cpp
Status ok;
ok = std::move(ok);
Status bad{Error::InvalidArgument, "Foo"};
bad = std::move(bad);
```